### PR TITLE
Added support for selecting an AWS profile in AwsOsduClient

### PR DESCRIPTION
**Pull request by pbradshawusc**

### CODE REVIEW

File: osdu/client/aws.py [hunk 1]
```diff
@@ -12,32 +12,48 @@ class AwsOsduClient(BaseOsduClient):
...
-            self._profile = None
 
 
     def get_tokens(self, password) -> None:
-        if self._profile:
-            aws_session = boto3.session.Session(profile_name=self._profile)
-            client = my_session.client('cognito-idp')
-        else:
-            client = boto3.client('cognito-idp')
+        session = boto3.Session(profile_name=self._profile)
```
Comment (pbradshawusc) [inline osdu/client/aws.py:53]: We only want to do this if the profile name does exist. To preserve use in the backend, we need to detect if the profile name is set and conditionally use `Session` or `client` to create the Cognito client. 

The reason is that we need to use a `Session` if we have a profile name as that is the construct that takes in a given AWS profile to create the session. However, on the backend where we are using IAM roles and not profiles/local creds, we cannot use a `Session` which will automatically try and use the default credentials. Instead we need to use the `client` directly so that it will pick up on the backend authentication. 

tl;dr - we need to check for no profile and use client in that case or our backend dataload scripts will break

Comment (puremcc) [reply osdu/client/aws.py:53]: OK, thanks for the explanation, and apologies as I should have proposed as a review comment instead of committing the change. I'm fine with sticking with your proposed approach, but let's sync up later to discuss. I'd like to better understand what is actually going on when you say '_use the client directly so that it will pick up on the backend authentication_'.

**Review by puremcc — APPROVED** [review]
